### PR TITLE
Loads metric states from state_dict

### PR DESCRIPTION
**Pull request by hlin09**
The persistent metric states are output in `state_dict`, but cannot be properly loaded through `load_state_dict()`. This PR adds handling to loading persisted metric states from `state_dict` and a unit test.

### CODE REVIEW

File: torchmetrics/metric.py [hunk 1]
```diff
@@ -324,6 +324,16 @@ def state_dict(self, destination=None, prefix="", keep_vars=False):
                 destination[prefix + key] = current_val
         return destination
 
+    def _load_from_state_dict(self, state_dict, prefix, local_metadata, strict,
+                              missing_keys, unexpected_keys, error_msgs):
```
Comment (ananthsub) [inline torchmetrics/metric.py:328]: add typehints here as well

**Review by ananthsub — COMMENTED** [review]
this lgtm - @SkafteNicki what do you think?

File: torchmetrics/metric.py (hunk 1 above)
Comment (SkafteNicki) [inline torchmetrics/metric.py:328]:
```suggestion
    def _load_from_state_dict(self, state_dict: dict, prefix: str, local_metadata: dict, strict: bool,
                              missing_keys: List[str], unexpected_keys: List[str], error_msgs: List[str]) -> None:
```

**Review by SkafteNicki — APPROVED** [review]
LGTM

**Review by Borda — APPROVED** [review]
